layerindex/views.py

```diff
@@ -944,13 +1464,493 @@ def get_context_data(self, **kwargs):
+    if not request.user.has_perm('layerindex.update_comparison_branch'):
+        raise PermissionDenied
+
+    from celery import uuid
+
+    cmd = None
+    for item in getattr(settings, 'COMPARISON_UPDATE', []):
+        if item['branch_name'] == branchobj.name:
+            cmd = item['update_command']
+            break
+    if not cmd:
+        raise Exception('No update command defined for branch %s' % branch)
+
+    task_id = uuid()
+    # Create this here first, because inside the task we don't have all of the required info
+    update = Update(task_id=task_id)
+    update.started = datetime.now()
+    update.triggered_by = request.user
+    update.save()
+
+    res = tasks.run_update_command.apply_async((branch, cmd), task_id=task_id)
+
+    return HttpResponseRedirect(reverse_lazy('task_status', kwargs={'task_id': task_id}))
+
+
+class TaskStatusView(TemplateView):
+    def get_context_data(self, **kwargs):
+        from celery.result import AsyncResult
+        context = super(TaskStatusView, self).get_context_data(**kwargs)
+        task_id = self.kwargs['task_id']
+        context['task_id'] = task_id
+        context['result'] = AsyncResult(task_id)
+        context['update'] = get_object_or_404(Update, task_id=task_id)
+        context['log_url'] = reverse_lazy('task_log', args=(task_id,))
         return context
+
+def task_log_view(request, task_id):
+    from celery.result import AsyncResult
+    if not request.user.is_authenticated:
+        raise PermissionDenied
+
+    if '/' in task_id:
+        # Block anything that looks like a path
+        raise Http404
+
+    result = AsyncResult(task_id)
+    start = int(request.GET.get('start', 0))
+    try:
+        f = open(os.path.join(settings.TASK_LOG_DIR, 'task_%s.log' % task_id), 'rb')
+    except FileNotFoundError:
+        raise Http404
+    try:
+        f.seek(start)
+        datastr = f.read()
+        origlen = len(datastr)
+        # Squash out CRs *within* the string (CRs at the start preserved)
+        datastr = re.sub(b'\n[^\n]+\r', b'\n', datastr)
+        # We need to escape this or else things that look like tags in the output
+        # will be interpreted as such by the browser
+        data = escape(datastr)
+        response = HttpResponse(data)
+        try:
+            ready = result.ready()
+        except ConnectionResetError:
+            # FIXME this shouldn't be happening so often, but ultimately we don't care -
+            # the frontend is polling so it'll likely succeed in a subsequent request
+            ready = False
+        if ready:
+            response['Task-Done'] = '1'
+            updateobj = get_object_or_404(Update, task_id=task_id)
+            response['Task-Duration'] = utils.timesince2(updateobj.started, updateobj.finished)
+            response['Task-Progress'] = 100
+            if result.info:
+                if isinstance(result.info, dict):
+                    response['Task-Result'] = result.info.get('retcode', None)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears you are using the `get()` method to retrieve the value for a key from a dictionary. If the key is not present, then `get()` returns the value that is passed in its optional second parameter. If if the key is not present, then `None` is the default return value. Therefore, there is no need to explicitly specify `None` in the second parameter.

[Learn more](https://docs.python.org/3/library/stdtypes.html#dict.get)